### PR TITLE
fix(docs): Don't hide `@alpha` APIs from website docs

### PR DESCRIPTION
**Pull request by Josmithr**
Previously, when `@alpha` was strictly reserved for APIs used by internal partners, we update the website build to hide such members from public documentation. Later, the definition of `@alpha` was extended to mean "things we don't want new adoption of", and public-facing APIs like SharedMap were tagged as `@alpha`. As a result, our public API documentation has been substantially broken for some time.

This PR fixes the situation by updating the build to once again generate API docs for `@alpha` items.

It also updates the alerts we generate for `@alpha` and `@beta` to more accurately represent the usage contracts (including instructions for how to import them). E.g.
image
image

### Remaining work
There are still a number of places where `@alpha` items result in "ALPHA" disclaimer text. This behavior is not currently configurable by the documenter library. Subsequent PRs will add flexibility in the library, and leverage it in the website build.
E.g.
image

### CODE REVIEW

**Comment by Josmithr** [comment]
@jason-ha @DLehenbauer we will need to keep the website in mind with any future tagging definition changes. When we migrate to using `@legacy` tags, we'll want to undo much of this / change related alerts.

File: docs/api-markdown-documenter/custom-renderers.js [hunk 1]
```diff
@@ -16,9 +16,9 @@ export function renderAlertNode(alertNode, writer, context) {
 	writer.ensureNewLine();
 
 	writer.writeLine(
-		`{{% callout ${alertNode.alertKind?.toLocaleLowerCase() ?? "note"} ${
+		`{{% callout ${alertNode.alertKind?.toLocaleLowerCase() ?? "note"} "${
```
Comment (Josmithr) [inline docs/api-markdown-documenter/custom-renderers.js:19]: This was a bug. At some point, usages of the AlertNode must have been removed. Possibly in response to this? This fixes it, and allows us to leverage Hugo callouts as intended.

**Review by alexvy86 — APPROVED** [review]

File: docs/api-markdown-documenter/alert-node.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ export const alertNodeType = "Alert";
  * @example Example rendering output (in Hugo Markdown)
  *
  * ```md
- * {{% callout TIP Unit tests are super useful! %}}
+ * {{% callout tip "Unit tests are super useful!" %}}
```
Comment (Josmithr) [inline docs/api-markdown-documenter/alert-node.js:22]:
```suggestion
 * {{% callout note "Unit tests are super useful!" %}}
```

File: docs/api-markdown-documenter/api-documentation-layout.js [hunk 3]
```diff
@@ -18,25 +20,49 @@ import { AlertNode } from "./alert-node.js";
...
+	const packageName = apiItem.getAssociatedPackage().displayName;
+	return new AlertNode(
+		[
+			new SpanNode([
+				new PlainTextNode("To use, import via "),
+				CodeSpanNode.createFromPlainText(`${packageName}/legacy`),
+				new PlainTextNode("."),
+			]),
+		],
+		/* alertKind: */ "note",
```
Comment (seanimam) [inline docs/api-markdown-documenter/api-documentation-layout.js:36]: discrepancy here with this being 'note' while for beta warning its a 'warning'. Just pointing out if incase this is not intended.

File: docs/api-markdown-documenter/api-documentation-layout.js [hunk 4]
```diff
@@ -18,25 +20,49 @@ import { AlertNode } from "./alert-node.js";
...
+		/* title: */ "This API is provided as-is for existing users, but is not recommended for new users.",
```
Comment (Josmithr) [inline docs/api-markdown-documenter/api-documentation-layout.js:37]: @kashms interested in your thoughts on this wording.

File: docs/api-markdown-documenter/api-documentation-layout.js [hunk 5]
```diff
@@ -18,25 +20,49 @@ import { AlertNode } from "./alert-node.js";
...
+	return new AlertNode(
+		[
+			new SpanNode([
+				new PlainTextNode("To use, import via "),
+				CodeSpanNode.createFromPlainText(`${packageName}/beta`),
+				new PlainTextNode("."),
+			]),
+		],
+		/* alertKind: */ "warning",
+		/* title: */ "This API is provided as a beta preview and may change without notice. Use at your own risk.",
```
Comment (Josmithr) [inline docs/api-markdown-documenter/api-documentation-layout.js:52]: @kashms interested in your thoughts on this wording.

File: docs/api-markdown-documenter/api-documentation-layout.js (hunk 3 above)
Comment (Josmithr) [reply docs/api-markdown-documenter/api-documentation-layout.js:36]: It is intentional. We make real support guarantees for our "legacy" APIs, and they _are_ intended for production use. But we don't make support guarantees for beta APIs. So I figured the distinction in alert level was merited.

File: docs/api-markdown-documenter/api-documentation-layout.js (hunk 4 above)
Comment (Josmithr) [reply docs/api-markdown-documenter/api-documentation-layout.js:37]: Also wondering if we should note the support contract here. Presumably: will offer support for bugs, but are not taking feature requests around it.

Comment (kashms) [reply docs/api-markdown-documenter/api-documentation-layout.js:37]: I think we should remove the `as-is` bit and keep the "This API is not recommended for new users" and link out to our FF.com documentation here for more information: https://fluidframework.com/docs/build/releases-and-apitags/#api-tags

File: docs/api-markdown-documenter/api-documentation-layout.js (hunk 5 above)
Comment (kashms) [reply docs/api-markdown-documenter/api-documentation-layout.js:52]: Same as the last comment. I would remove the "Use at your own risk." part too :)

**Review by kashms — COMMENTED** [review]
Left a couple of comments to suggest small text changes + adding a link to our public docs regarding API docs here: https://fluidframework.com/docs/build/releases-and-apitags/#api-tags
Better to keep the details centralized

**Review by kashms — APPROVED** [review]
LGTM 🚢

**Pull request merged**